src/equations/shallow_water_2d.jl

```diff
@@ -0,0 +1,564 @@
+  p_avg  = 0.5 * equations.gravity * h2_avg
+  v_dot_n_avg = 0.5 * (v_dot_n_ll + v_dot_n_rr)
+
+  # Calculate fluxes depending on normal_direction
+  f1 = h_avg * v_dot_n_avg
+  f2 = f1 * v1_avg + p_avg * normal_direction[1]
+  f3 = f1 * v2_avg + p_avg * normal_direction[2]
+
+  return SVector(f1, f2, f3, zero(eltype(u_ll)))
+end
+
+
+"""
+    flux_wintermeyer_etal(u_ll, u_rr, normal_direction,
+                          equations::ShallowWaterEquations2D)
+
+Total energy conservative (mathematical entropy for shallow water equations) split form.
+When the bottom topography is nonzero this scheme will be well-balanced when used as a `volume_flux`.
+The `surface_flux` should still use, e.g., [`flux_fjordholm_etal`](@ref).
+
+Further details are available in Theorem 1 of the paper:
+- Niklas Wintermeyer, Andrew R. Winters, Gregor J. Gassner and David A. Kopriva (2017)
+  An entropy stable nodal discontinuous Galerkin method for the two dimensional
+  shallow water equations on unstructured curvilinear meshes with discontinuous bathymetry
+  [DOI: 10.1016/j.jcp.2017.03.036](https://doi.org/10.1016/j.jcp.2017.03.036)
+"""
+@inline function flux_wintermeyer_etal(u_ll, u_rr, normal_direction::AbstractVector, equations::ShallowWaterEquations2D)
+  # Unpack left and right state
+  h_ll, h_v1_ll, h_v2_ll, _ = u_ll
+  h_rr, h_v1_rr, h_v2_rr, _ = u_rr
+
+  # Get the velocities on either side
+  v1_ll, v2_ll = velocity(u_ll, equations)
+  v1_rr, v2_rr = velocity(u_rr, equations)
+
+  # Average each factor of products in flux
+  h_v1_avg = 0.5 * (h_v1_ll + h_v1_rr )
+  h_v2_avg = 0.5 * (h_v2_ll + h_v2_rr )
+  v1_avg   = 0.5 * (v1_ll   + v1_rr   )
+  v2_avg   = 0.5 * (v2_ll   + v2_rr   )
+  p_avg    = 0.5 * equations.gravity * h_ll * h_rr
+
+  # Calculate fluxes depending on normal_direction
+  f1 = h_v1_avg * normal_direction[1] + h_v2_avg * normal_direction[2]
+  f2 = f1 * v1_avg + p_avg * normal_direction[1]
+  f3 = f1 * v2_avg + p_avg * normal_direction[2]
+
+  return SVector(f1, f2, f3, zero(eltype(u_ll)))
+end
+
+
+# Calculate maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed of sound
+# TODO: This doesn't really use the `orientation` - should it?
+@inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer, equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  v1_ll, v2_ll = velocity(u_ll, equations)
+  h_rr = u_rr[1]
+  v1_rr, v2_rr = velocity(u_rr, equations)
+
+  # Calculate velocity magnitude and wave celerity on the left and right
+  v_mag_ll = sqrt(v1_ll^2 + v2_ll^2)
+  c_ll = sqrt(equations.gravity * h_ll)
+  v_mag_rr = sqrt(v1_rr^2 + v2_rr^2)
+  c_rr = sqrt(equations.gravity * h_rr)
+
+  λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
+end
+
+@inline function max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, equations::ShallowWaterEquations2D)
+  return max_abs_speed_naive(u_ll, u_rr, 0, equations) * norm(normal_direction)
```

Comment:
Maybe add a comment why the `0` is passed here?

Reply:
Sure I can add something. I think an analogous routine exists in most (if not all) 2d equation files. Should the comment I come up with also be put in those files?

Reply:
Only if you think it helps. I just saw it for the first time and thought "huh, why is this passed a zero", but maybe it's just me. Otherwise, feel free to resolve this without further action.